packages/testnet-faucets/src/ts/handler.ts

```diff
@@ -62,37 +77,83 @@ export class Handler {
-        if (_.isUndefined(recipientAddress) || !this._isValidEthereumAddress(recipientAddress)) {
-            res.status(400).send('INVALID_RECIPIENT_ADDRESS');
-            return;
-        }
-        const networkId = _.get(req.query, 'networkId', DEFAULT_NETWORK_ID);
-        const requestQueue = _.get(requestQueueByNetworkId, networkId);
+        const requestQueue = _.get(requestQueueByNetworkId, req.networkId);
         if (_.isUndefined(requestQueue)) {
-            res.status(400).send('INVALID_NETWORK_ID');
+            res.status(400).send('UNSUPPORTED_NETWORK_ID');
             return;
         }
-        const lowerCaseRecipientAddress = recipientAddress.toLowerCase();
-        const didAddToQueue = requestQueue.add(lowerCaseRecipientAddress);
+        const didAddToQueue = requestQueue.add(req.recipientAddress);
         if (!didAddToQueue) {
             res.status(503).send('QUEUE_IS_FULL');
             return;
         }
-        utils.consoleLog(`Added ${lowerCaseRecipientAddress} to queue: ${queueType} networkId: ${networkId}`);
+        utils.consoleLog(`Added ${req.recipientAddress} to queue: ${requestedAssetType} networkId: ${req.networkId}`);
         res.status(200).end();
     }
+    private async _dispenseOrder(req: express.Request, res: express.Response, requestedAssetType: RequestedAssetType) {
+        const zeroEx = _.get(this._zeroExByNetworkId, req.networkId);
+        if (_.isUndefined(zeroEx)) {
+            res.status(400).send('UNSUPPORTED_NETWORK_ID');
+            return;
+        }
+        res.setHeader('Content-Type', 'application/json');
+        const makerTokenAddress = await zeroEx.tokenRegistry.getTokenAddressBySymbolIfExistsAsync(requestedAssetType);
+        if (_.isUndefined(makerTokenAddress)) {
+            res.status(400).send('INVALID_TOKEN');
+            return;
+        }
+        const takerTokenSymbol =
+            requestedAssetType === RequestedAssetType.WETH ? RequestedAssetType.ZRX : RequestedAssetType.WETH;
+        const takerTokenAddress = await zeroEx.tokenRegistry.getTokenAddressBySymbolIfExistsAsync(takerTokenSymbol);
+        if (_.isUndefined(takerTokenAddress)) {
+            res.status(400).send('INVALID_TOKEN');
+            return;
+        }
+        const makerTokenAmount = new BigNumber(0.1);
+        const takerTokenAmount = new BigNumber(0.1);
+        const order: Order = {
+            maker: configs.DISPENSER_ADDRESS,
+            taker: req.recipientAddress,
+            makerFee: new BigNumber(0),
+            takerFee: new BigNumber(0),
+            makerTokenAmount,
+            takerTokenAmount,
+            makerTokenAddress,
+            takerTokenAddress,
+            salt: ZeroEx.generatePseudoRandomSalt(),
+            exchangeContractAddress: zeroEx.exchange.getContractAddress(),
+            feeRecipient: ZeroEx.NULL_ADDRESS,
+            expirationUnixTimestampSec: new BigNumber(Date.now() + TWO_DAYS_IN_MS),
```

Comment:
Let's make this longer. Perhaps 5 days? Otherwise these might expire by the end of the hackathon, right around demo time 🌮